fhir_zulip_nlp/fhir_zulip_nlp.py

```diff
@@ -206,13 +216,16 @@ def create_report2(
             df_kw = df_kw.sort_values(['timestamp'])  # oldest first
             threads: List[str] = list(df_kw['subject'].unique())
             # Average thread length
+            # TODO: Refactor to pandas?
             tot_thread_len = 0
             for thread in range(len(threads)):
                 df_thread = df_kw[df_kw['subject'] == threads[thread]]
                 z = (list(df_thread['timestamp'])[-1] - list(df_thread['timestamp'])[0]) / seconds_per_day
                 tot_thread_len += float(f'{z:.1f}')
             avg_thread_len = round(tot_thread_len / len(threads), 3)
             # Outliers
+            # TODO: Refactor to pandas?
```

Comment:
@rohaher I haven't yet refactored to improve performance as you asked, but I added a TODO here.